tests/integration/networks/evm/contract_interaction.rs

```diff
@@ -0,0 +1,228 @@
+    info!(
+        tx_id = %tx_response.id,
+        status = %tx_response.status,
+        "Transaction submitted"
+    );
+
+    let receipt_config = ReceiptConfig::from_network(network)?;
+    debug!(
+        poll_interval_ms = receipt_config.poll_interval_ms,
+        max_wait_ms = receipt_config.max_wait_ms,
+        "Waiting for confirmation"
+    );
+
+    if let Err(e) = wait_for_receipt(&client, &relayer.id, &tx_response.id, &receipt_config).await {
+        if let Ok(failed_tx) = client.get_transaction(&relayer.id, &tx_response.id).await {
+            error!(
+                status = %failed_tx.status,
+                hash = ?failed_tx.hash,
+                error = ?failed_tx.error,
+                "Transaction failed"
+            );
+        }
+        return Err(e);
+    }
+
+    let final_tx = client.get_transaction(&relayer.id, &tx_response.id).await?;
+    info!(
+        tx_id = %final_tx.id,
+        hash = ?final_tx.hash,
+        status = %final_tx.status,
+        "Transaction confirmed"
+    );
+
+    if final_tx.status != "confirmed" && final_tx.status != "mined" {
+        return Err(eyre::eyre!(
+            "Transaction status should be confirmed or mined, got: {}",
+            final_tx.status
+        ));
+    }
+
+    if final_tx.hash.is_none() {
+        return Err(eyre::eyre!("Confirmed transaction should have a hash"));
+    }
+
+    info!("Contract interaction test completed successfully");
+    Ok(())
+}
+
+/// Test SimpleStorage contract interaction on all selected EVM networks
+#[tokio::test]
+#[ignore = "Requires running relayer, funded signer, and deployed contract"]
+#[serial]
+async fn test_evm_contract_interaction() {
+    init_test_logging();
+
+    let _span = info_span!("test_evm_contract_interaction").entered();
+
+    let networks = get_test_networks().expect("Failed to get test networks");
+
+    if networks.is_empty() {
+        panic!("No networks selected for testing");
+    }
+
+    let registry = TestRegistry::load().expect("Failed to load test registry");
+
+    let eligible_networks: Vec<String> = networks
+        .into_iter()
+        .filter(|network| {
+            let is_evm = registry
+                .get_network(network)
+                .map(|config| config.network_type == "evm")
+                .unwrap_or(false);
+
+            let has_contract = registry
+                .has_real_contract(network, "simple_storage")
+                .unwrap_or(false);
+
+            is_evm && has_contract
+        })
+        .collect();
+
+    if eligible_networks.is_empty() {
+        info!("No EVM networks with deployed SimpleStorage contract, skipping test");
+        return;
+    }
+
+    info!(
+        count = eligible_networks.len(),
+        networks = ?eligible_networks,
+        "Testing EVM networks with SimpleStorage"
+    );
+
+    let mut failures = Vec::new();
+
+    // Run test for each eligible network
+    for network in &eligible_networks {
```

Comment:
This logic is repetitive. We should optimise it by defining once and using where needed. Either util method or some macro.